exporters/jaeger/lib/opentelemetry/exporters/jaeger/exporter.rb

```diff
@@ -79,43 +80,9 @@ def encoded_batches(span_data)
           SUCCESS
         end
 
-        def encoded_span(span_data) # rubocop:disable Metrics/AbcSize
-          start_time = (span_data.start_timestamp.to_f * 1_000_000).to_i
-          duration = (span_data.end_timestamp.to_f * 1_000_000).to_i - start_time
-
-          Thrift::Span.new(
-            'traceIdLow' => int64(span_data.trace_id[16, 16]),
-            'traceIdHigh' => int64(span_data.trace_id[0, 16]),
-            'spanId' => int64(span_data.span_id),
-            'parentSpanId' => int64(span_data.parent_span_id),
-            'operationName' => span_data.name,
-            'references' => encoded_references(span_data.links),
-            'flags' => span_data.trace_flags.sampled? ? 1 : 0,
-            'startTime' => start_time,
-            'duration' => duration,
-            'tags' => encoded_tags(span_data.attributes) + encoded_status(span_data.status) + encoded_kind(span_data.kind),
-            'logs' => encoded_logs(span_data.events)
-          )
-        end
-
-        def encoded_logs(events)
-          events&.map do |event|
-            Thrift::Log.new(
-              'timestamp' => (event.timestamp.to_f * 1_000_000).to_i,
-              'fields' => encoded_tags(event.attributes) + encoded_tags('name' => event.name)
-            )
-          end
-        end
-
-        def encoded_references(links)
-          links&.map do |link|
-            Thrift::SpanRef.new(
-              'refType' => Thrift::SpanRefType::CHILD_OF,
-              'traceIdLow' => int64(link.context.trace_id[16, 16]),
-              'traceIdHigh' => int64(link.context.trace_id[0, 16]),
-              'spanId' => int64(link.context.span_id)
-            )
-          end
+        def encoded_span(span_data)
+          @span_encoder ||= SpanEncoder.new
```

Comment:
I don't think there's any value in lazily initializing the encoder. I'd just initialize it in `initialize`. At that point, you might also want to inline this method at its only callsite.

Reply:
I suppose there's the miniscule optimization of not initializing the encoder if no spans are ever sent, but that's pretty minimal. I'll change it now.